lib/fizzy/parser.cpp

```diff
@@ -1,10 +1,29 @@
 #include "parser.hpp"
+#include "leb128.hpp"
+#include <cassert>
 
 namespace fizzy
 {
-module parse(bytes_view bin)
+module parse(bytes_view input)
 {
-    (void)bin;
-    return {};
+    assert(input.substr(0, wasm_prefix.size()) == wasm_prefix);
+
+    input.remove_prefix(wasm_prefix.size());
+
+    module mod;
+    for (auto it = input.begin(); it != input.end();)
+    {
+        const auto id = static_cast<sectionid>(*it++);
+        const auto [size, new_pos] = leb128u_decode<uint32_t>(it);
+        it = new_pos;
+        switch (id)
+        {
+        default:
+            break;
+        }
+        it += size;
```

Comment:
I think this should move into the `default` case.

Reply:
This is for all sections, also the "implemented" ones.

Reply:
For parsed section we should check that parser consumed the expected number of bytes

Reply:
Can it consume less?

Reply:
I will add the more restrictive check.